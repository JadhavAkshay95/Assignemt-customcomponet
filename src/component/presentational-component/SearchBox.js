import React, { useEffect, useState } from "react";

const SearchBox = ({ filterData }) => {
  const [search, setSearch] = useState();

  useEffect(() => {
    filterData(search);
  }, [search]);

  const searchHandler = ($event) => {
    setSearch($event.target.value);
  };

  return (
    <>
      <div>
        <input
          placeholder="search by name"
          value={search}
          onChange={searchHandler}
        />
      </div>
    </>
  );
};
export default SearchBox;
