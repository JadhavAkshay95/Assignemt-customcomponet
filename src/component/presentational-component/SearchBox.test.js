import { shallow } from "enzyme";
import SearchBox from "./SearchBox";
import React from "react";

describe("Search Box Component", () => {
  it("Should render Searc hBox component", () => {
    const wrapper = shallow(<SearchBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
