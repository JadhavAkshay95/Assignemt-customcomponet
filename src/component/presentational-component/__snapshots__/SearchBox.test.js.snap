// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box Component Should render Searc hBox component 1`] = `
<Fragment>
  <div>
    <input
      onChange={[Function]}
      placeholder="search by name"
    />
  </div>
</Fragment>
`;
